HW04.Rmd

```diff
@@ -1,126 +1,343 @@
-
-```{r police}
-
-```
-
-Think about performing stats such as a t.test to see if differences you are plotting are statistically significant
-
-```{r stats?}
-
-
-```
-
-
-```{r police_2}
-
-```
-
-```{r stats_2?}
-
-
-```
-
-
-Finally, all together now:
-```{r police_3}
-
-```
-
-
-## Session info
-
-```{r}
-# always good to have this for reproducibility purposes
-devtools::session_info()
-```
-
+---
+title: "Exploring Gun Deaths in America"
+output: github_document
+---
+
+## Objectives
+* To explore gun death data curated by FiveThirtyEight [original data available here](https://github.com/fivethirtyeight/guns-data) by
+    * Month
+    * Intent
+    * Age and sex
+    * Education
+    * Season
+    * Intent by race
+    * Police-relatedness 
+
+## Read in the data
+
+The gun_deaths.csv is provided in the repository. 
+```{r load-data, message=FALSE, warning=FALSE}
+library(tidyverse)    # load tidyverse packages, including ggplot2
+library(knitr) # load knitr package so that I can use kable() function
+
+# read in data and define as gun_deaths
+gun_deaths <- read.csv("gun_deaths.csv")
+
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+
+# generate new dataframe for this task
+gun_deaths_summary1 <- gun_deaths %>%
+    select(month) %>%
+    group_by(month) %>%
+    count(month) %>%
+    rename(deaths_per_month = n)
+
+# print results as a table using kable()
+kable(gun_deaths_summary1, format = "markdown", align = 'c', col.names = c("Month", "Deaths per Month"))
+
+
+```
+
+**If we further want to divide the number of gun deaths per month for each given year:**
+
+```{r month_2, warning=FALSE}
+
+# generate new dataframe, almost the same as above but also selecting and grouping by year
+gun_deaths_summary2 <- gun_deaths %>%
+    select(year, month) %>%
+    group_by(year, month) %>%
+    count(month) %>%
+    rename(deaths_per_month = n)
+
+# print results as a table using kable()
+kable(gun_deaths_summary2, format = "markdown", align = 'c', col.names = c("Year", "Month", "Deaths per Month"))
+
+
+```
+
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+
+# generate bar graph using the dataframe generated above
+ggplot(data = gun_deaths_summary1, aes(x = factor(month), y = deaths_per_month)) + # make sure month is plotted as factor so that we can convert to abbreviated month name
+    geom_col() +
+    scale_x_discrete(labels=month.abb) + # change x-axis labels to abbreviated month names instead of numbers
```

Comment:
This was such a clean way of changing the axis labels! I went through the trouble of modifying the dataset _before_ creating this plot, and this is a much simpler (and easier to follow) way of doing it. I'm definitely taking note of this method.

Reply:
I also ended up manually specifying abbreviations for each month and I saw lots of people suggesting this solution in similar situations. Yet this is so much cleaner and does the job without modifying the dataframe. Great!